### PR TITLE
[ci] Update NVIDIA driver packages to v530 in docker images

### PR DESCRIPTION
**Pull request by antiagainst**
These driver packages are shared between CUDA and Vulkan. This is a preparation step for updating CUDA to 12.1.1, which requires the driver version to be >=525.60.13.

Progress towards https://github.com/openxla/iree/issues/13830

### CODE REVIEW

**Comment by antiagainst** [comment]
Yeah, as expected, we'd need to update the host VM's drivers to v530 too: https://github.com/openxla/iree/actions/runs/5151975373/jobs/9277728266#step:6:176.

**Comment by antiagainst** [comment]
It looks we have some improvements due to updating the drivers. Nice. :)

File: build_tools/docker/dockerfiles/swiftshader-bleeding-edge.Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-FROM gcr.io/iree-oss/base-bleeding-edge@sha256:3ea6d37221a452058a7f5a5c25b4f8a82625e4b98c9e638ebdf19bb21917e6fd AS install-swiftshader
```
Comment (GMNGeoffrey) [inline build_tools/docker/dockerfiles/swiftshader-bleeding-edge.Dockerfile]: Why did this get updated? I don't see any changes to base here.

File: build_tools/docker/dockerfiles/nvidia-bleeding-edge.Dockerfile
Comment (pzread) [inline build_tools/docker/dockerfiles/nvidia-bleeding-edge.Dockerfile:1]: Looks like we still install CUDA SDK 11.8 at line 65. Are we planning to update that later?

File: build_tools/docker/dockerfiles/nvidia.Dockerfile
Comment (pzread) [inline build_tools/docker/dockerfiles/nvidia.Dockerfile:1]: Looks like we still install CUDA SDK 11.6 at line 54. Are we planning to update that later?

File: build_tools/docker/dockerfiles/nvidia-bleeding-edge.Dockerfile
Comment (antiagainst) [reply build_tools/docker/dockerfiles/nvidia-bleeding-edge.Dockerfile:1]: Yes, that would happen later in https://github.com/openxla/iree/pull/13875.

**Review by pzread — APPROVED** [review]

File: build_tools/docker/dockerfiles/swiftshader-bleeding-edge.Dockerfile (hunk 1 above)
Comment (antiagainst) [reply build_tools/docker/dockerfiles/swiftshader-bleeding-edge.Dockerfile]: Good catch! I don't know actually.. Maybe messed up with some other docker updates.. Reverted.

File: build_tools/github_actions/runner/gcp/image_setup.sh [hunk 2]
```diff
@@ -314,8 +314,8 @@ EOF
           bash -c "${script_dir}/check_cuda.sh && ${script_dir}/check_vulkan.sh"
     }
 
-    check_docker gcr.io/iree-oss/nvidia@sha256:f8c6680f435615256c9df7521d41ad49f13bbb34e2dd16a271a620e9dc8637e7
-    check_docker gcr.io/iree-oss/frontends-nvidia@sha256:808f57962267eb1752c3200348421a22eca1770c737922ee2b933585d16f014d
+    check_docker gcr.io/iree-oss/nvidia@sha256:4c113db4f62cf19aa6aebdccc86e3b6407edee075f6ffc0ffe732e115a4ae7e5
+    check_docker gcr.io/iree-oss/frontends-nvidia@sha256:bc217d1a226f778e44874b989d6ead3a3085f04643a40aa301f5d30661356d07
```
Comment (pzread) [inline build_tools/github_actions/runner/gcp/image_setup.sh:318]: (Unrelated to this PR): I just noticed that we didn't check `nvidia-bleeding-edge` in the `image_setup.sh` script

**Pull request merged**